### PR TITLE
Crazy4pi314/quantumdevwme

### PR DESCRIPTION
**Pull request by crazy4pi314**
This PR completes #15 and #14. It added some new types and helpers for classical data set management, as well as some style consistency and naming updates.

### CODE REVIEW

File: samples/BucketBrigade/BucketBrigadeSample.qs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    open Microsoft.Quantum.Arithmetic;
+    open Microsoft.Quantum.Canon;
+    open Microsoft.Quantum.Convert;
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Measurement;
+    
+    open Qram;
+
+    @EntryPoint()
+    operation QromSample(queryAddress : Int) : Int {
```
Comment (glassnotes) [inline samples/BucketBrigade/BucketBrigadeSample.qs]: Qram?

File: samples/Qrom/QromSample.qs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿namespace implicitQRAM {
+﻿namespace QromSample {
```
Comment (glassnotes) [inline samples/Qrom/QromSample.qs:1]: This file has ImplicitQRAM some places and QROM in others - but let's take care of this when we get to the QROMs.

File: src/BucketBrigade.qs [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                X(Head(auxRegister));
+                ApplyAddressFanout(addressRegister, auxRegister);
+            }
+            apply {
+                ReadoutMemory(memoryRegister, auxRegister, targetRegister);
+            }
+        } 
+    }
+
+    operation ReadoutMemory(
```
Comment (glassnotes) [inline src/BucketBrigade.qs]: This operation is missing docs; also should it maybe be changed to `internal`? Is it used outside of BucketBrigadeRead?

**Review by glassnotes — APPROVED** [review]
I made a couple small comments, but otherwise everything looks great! Thanks :+1:

**Comment by crazy4pi314** [comment]
Gonna merge it then, thanks for the review! <3

**Pull request merged**